### PR TITLE
chore(codemod): interactive cli

### PR DESCRIPTION
**Pull request by talkor**
https://monday.monday.com/boards/3532714909/pulses/96a44868-fa10-c4e0-4bd5-344eb900a011

### CODE REVIEW

**Review by YossiSaadi — COMMENTED** [review]
Will continue to go over it later

File: packages/codemod/package.json [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
   "name": "@vibe/codemod",
-  "version": "0.0.3",
-  "type": "module",
+  "version": "0.0.5",
   "description": "Vibe's component library migration tool",
   "repository": "git+https://github.com/mondaycom/vibe.git",
   "author": "monday.com",
   "license": "MIT",
-  "bin": "./bin/vibe-codemod.js",
+  "bin": "./dist/cli.js",
```
[the remark this answers is not in the document]
Comment (talkor) [reply packages/codemod/package.json]: Cha

File: packages/codemod/bin/vibe-codemod.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+      const child = spawn(
+        "node",
+        [
+          require.resolve("jscodeshift/bin/jscodeshift"),
+          `--extensions=${extensions.join(",")}`,
+          "--ignore-pattern=node_modules",
+          "--ignore-pattern=**/*.d.ts",
+          `--verbose=${verbosityLevel}`,
+          "--max-workers=8",
+          "--no-babel",
```
Comment (talkor) [inline packages/codemod/bin/vibe-codemod.ts:152]: https://github.com/facebook/jscodeshift/issues/294

File: packages/codemod/bin/vibe-codemod.ts [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      });
+    });
+  }
+
+  const transformationFiles: string[] = globby.sync(`${transformationsDir}/*.js`, {
+    ignore: ["node_modules/**", "**/*.d.ts"]
+  });
+
+  if (transformationFiles.length === 0) {
+    console.error(chalk.red("No transformation files found. Please check the directory path."));
```
Comment (YossiSaadi) [inline packages/codemod/bin/vibe-codemod.ts:203]: maybe "please check the migration name supplied?"
as this scenario shouldn't happen unless specified with wrong migration name?

File: packages/codemod/bin/vibe-codemod.ts [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+    process.exit(1);
+  }
+
+  const filesToProcessLast = [
+    resolve(transformationsDir, "type-imports-migration.js"),
+    resolve(transformationsDir, "packages-rename-migration.js")
```
Comment (YossiSaadi) [inline packages/codemod/bin/vibe-codemod.ts:209]: .js?

**Review by YossiSaadi — APPROVED** [review]
I'm really really excited, I swear 🤩

File: packages/codemod/bin/vibe-codemod.ts (hunk 4 above)
Comment (talkor) [reply packages/codemod/bin/vibe-codemod.ts:209]: Yes we deploy js, it runs tsc on build

Comment (YossiSaadi) [reply packages/codemod/bin/vibe-codemod.ts:209]: yes, but do you really need to declare the extension?

Comment (talkor) [reply packages/codemod/bin/vibe-codemod.ts:209]: Oh yeah, will remove, unneeded

Comment (talkor) [reply packages/codemod/bin/vibe-codemod.ts:209]: Turns out it's needed, keeping

**Pull request merged**